File: Client/src/main/java/com/rspsi/editor/model/TileCoordinate.java
package com.rspsi.editor.model;

/**
 * Legacy local-document tile coordinate.
 *
 * @deprecated New viewport/navigation code must use WorldTile and new
 * document-facing code must use LocalTile. This compatibility type remains
 * temporarily for command/history/selection contracts while they migrate.
 */
@Deprecated
public record TileCoordinate(int plane, int x, int y) {
    public TileCoordinate {
        if (plane < 0 || x < 0 || y < 0) {
            throw new IllegalArgumentException("Tile coordinates cannot be negative");
        }
    }
}

File: Client/src/main/kotlin/com/rspsi/editor/model/TileCoordinate.kt
package com.rspsi.editor.model

/**
 * Legacy document-local tile coordinate retained only for compatibility.
 *
 * New document-facing code should use [LocalTile]. Absolute OSRS coordinates must use
 * [WorldTile]. This type remains while command/history/selection contracts finish migrating,
 * so its Java record ABI, deprecation metadata, and validation behavior are intentionally preserved.
 */
@kotlin.Deprecated("Use LocalTile for document coordinates or WorldTile for absolute coordinates")
@java.lang.Deprecated
@JvmRecord
data class TileCoordinate(
    val plane: Int,
    val x: Int,
    val y: Int,
) {
    init {
        if (plane < 0 || x < 0 || y < 0) {
            throw IllegalArgumentException("Tile coordinates cannot be negative")
        }
    }

    override fun toString(): String =
        "TileCoordinate[plane=$plane, x=$x, y=$y]"
}
